clients/src/main/java/org/apache/kafka/common/requests/DeleteTopicsRequest.java

```diff
@@ -39,8 +45,25 @@ public Builder(DeleteTopicsRequestData data) {
 
         @Override
         public DeleteTopicsRequest build(short version) {
+            if (version >= 6 && data.topicNames().size() != 0) {
+                data.setTopics(groupByTopic(data.topicNames()));
+            } else if (version >= 6) {
+                for (DeleteTopicState topic : data.topics()) {
+                    if (topic.name().equals("")) {
```

Comment:
if topic.name() == null, a NEP will throw here.

Reply:
Good point. I think I was assuming the default is the empty string, but someone could set the data field to null.